### DISCUSSION TITLE
PLEASE document somewhere that `npm install` will format package.json after execution

### DISCUSSION THREAD

**Discussion by albertodiazdorado** [npm/feedback#358] [CLI] [answered]
Why did I have to figure out by try-and-error **who, when** was formatting my `package.json`?

`yarn` also does this. I don't know why `npm` or `yarn` think that they are cool enough to go ahead and format my files to their hearts content, but alas, let it be like that.

In any case, I would love it if this behavior was documented somewhere. I don't want to open a bug ticket in the main repo just for this, so I am opening a discussion here.

EDIT: Just to be clear. I know that `npm install` add/removes packages from the `package.json`. What I did not know is that **it also formats the `package.json` using the formatting rules of its preference**. That is the part of the documentation that is missing IMO

**Comment by MylesBorins** [comment, marked as answer]
This is documented in the `npm install` documentation

> npm install saves any specified packages into dependencies by default. 

https://docs.npmjs.com/cli/v7/commands/npm-install

**Reply by albertodiazdorado** [reply]
Hi @MylesBorins ! Sure, it saves any packages into the `package.json` - but it also **formats** the `package.json`. It is not said anywhere (as far as I know) that `npm install`, in addition to adding packages to the `package.json`, will also format the file.

**Reply by MylesBorins** [reply]
@darcyclarke has offered to help get the docs updated to reflect this

**Reply by albertodiazdorado** [reply]
Ok @MylesBorins , thank you for the feedback

**Comment by albertodiazdorado** [comment]
Hi?
